src/textual/_compositor.py

```diff
@@ -877,15 +877,13 @@ def cuts(self) -> list[list[int]]:
             return self._cuts
 
         width, height = self.size
-        screen_region = self.size.region
         cuts = [[0, width] for _ in range(height)]
 
         intersection = Region.intersection
         extend = list.extend
 
         for region, clip in self.visible_widgets.values():
-            region = intersection(region, clip)
-            if region and (region in screen_region):
+            if region := intersection(region, clip):
```

Comment:
`visible_widgets` is already limited to the visible area, so no need for another `region in screen_region` check.